src/courses/course.service.ts

```diff
@@ -43,17 +43,28 @@ export class CourseService {
     }
     async enrollCourse(enrollData: EnrollInput) {
         try {
-            const course = await this.courseModel.findOneAndUpdate({ id: enrollData.courseID }, { $push: { enrolledUsers: enrollData.userID }, $inc: { enrolled: 1 } });
-            return course;
+            const course = await this.courseModel.findOne({ id: enrollData.courseID });
+            const isEnroll = course.enrolledUsers.includes(enrollData.userID)
```

Comment:
change line 46,47 with this one const isEnroll = await this.checkUserIsEnrolled(enrollData)